plugins/inputs/nats/nats.go

```diff
@@ -0,0 +1,104 @@
+package nats
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"encoding/json"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	gnatsd "github.com/nats-io/gnatsd/server"
+)
+
+type Nats struct {
+	Server          string
+	ResponseTimeout internal.Duration
+}
+
+var sampleConfig = `
+  ## The address of the monitoring endpoint of the NATS server
+  server = "http://localhost:1337"
+
+  ## Maximum time to receive response
+  # response_timeout = "5s"
+`
+
+func (n *Nats) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *Nats) Description() string {
+	return "Provides metrics about the state of a NATS server"
+}
+
+func (n *Nats) Gather(acc telegraf.Accumulator) error {
+	url, err := url.Parse(n.Server)
+	if err != nil {
+		return err
+	}
+	url.Path = path.Join(url.Path, "varz")
+
+	client := n.createHTTPClient()
```

Comment:
Only create a new http client once and reuse each gather.

Reply:
Done